plasmapy/physics/parameters.py

```diff
@@ -178,12 +179,13 @@ def Alfven_speed(B, density, ion="p"):
     except Exception:
         raise ValueError("Unable to find Alfven speed")
 
-    _check_relativistic(V_A, 'Alfven_speed')
-
     return V_A
 
 
-def ion_sound_speed(*ignore, T_e=0*units.K, T_i=0*units.K,
+@check_relativistic
+@check_quantity("T_e", can_be_negative=False)
+@check_quantity("T_i", can_be_negative=False)
```

Comment:
In case 2, I'm not immediately sure when those types would be checked (at each runtime call? haven't used that feature before), but what would happen if someone leaves the type off? Does it miss validation in that case? I think I prefer case 1, so that there's a single point of truth for all the validation info.